proxy/envoy/resources.go

```diff
@@ -211,40 +213,72 @@ type AccessLog struct {
 	Filter string `json:"filter,omitempty"`
 }
 
-// NetworkFilterConfig definition
-type NetworkFilterConfig struct {
-	CodecType         string       `json:"codec_type"`
-	StatPrefix        string       `json:"stat_prefix"`
-	GenerateRequestID bool         `json:"generate_request_id,omitempty"`
-	RouteConfig       *RouteConfig `json:"route_config"`
-	Filters           []Filter     `json:"filters"`
-	AccessLog         []AccessLog  `json:"access_log"`
-	Cluster           string       `json:"cluster,omitempty"`
+// HTTPFilterConfig definition
+type HTTPFilterConfig struct {
+	CodecType         string           `json:"codec_type"`
+	StatPrefix        string           `json:"stat_prefix"`
+	GenerateRequestID bool             `json:"generate_request_id,omitempty"`
+	RouteConfig       *HTTPRouteConfig `json:"route_config"`
+	Filters           []HTTPFilter     `json:"filters"`
+	AccessLog         []AccessLog      `json:"access_log"`
+}
+
+// TCPRoute definition
+type TCPRoute struct {
+	Cluster           string   `json:"cluster"`
+	DestinationIPList []string `json:"destination_ip_list,omitempty"`
+	DestinationPorts  []string `json:"destination_ports,omitempty"`
+	SourceIPList      []string `json:"source_ip_list,omitempty"`
+	SourcePorts       []string `json:"source_ports,omitempty"`
+}
+
+// TCPRouteConfigs provides routes by port
+type TCPRouteConfigs map[int]*TCPRouteConfig
+
+// TCPRouteConfig (or generalize as RouteConfig or L4RouteConfig for TCP/UDP?)
+type TCPRouteConfig struct {
+	Routes []TCPRoute `json:"routes"`
```

Comment:
The stat_prefix is part of the [TCPProxyFilterConfig](https://lyft.github.io/envoy/docs/configuration/network_filters/tcp_proxy_filter.html#config-network-filters-tcp-proxy-route-config). The [TCPRouteConfig](https://lyft.github.io/envoy/docs/configuration/network_filters/tcp_proxy_filter.html#config-network-filters-tcp-proxy-route-config) is just an array of TCPRoutes.